persistence/genre_repository.go

```diff
@@ -36,9 +38,17 @@ func (r *genreRepository) GetAll(opt ...model.QueryOptions) (model.Genres, error
 }
 
 func (r *genreRepository) Put(m *model.Genre) error {
-	id, err := r.put(m.ID, m)
-	m.ID = id
-	return err
+	if m.ID == "" {
+		m.ID = uuid.NewString()
+	}
+
+	return r.ormer.Raw(`
+INSERT INTO genre(id, name)
+VALUES(?, ?)
+ON CONFLICT (name) DO UPDATE
+	SET name=excluded.name
+RETURNING id
+`, m.ID, m.Name).QueryRow(&m.ID)
```

Comment:
Probably worth mentioning - if the approach is correct, but not the raw SQL, it's easy enough to a `Get()` and then only `Put()` if it doesn't exist.